tdp_server/services/runner.py

```diff
@@ -25,91 +17,42 @@ class StillRunningException(Exception):
 class RunnerService:
     def __init__(
         self,
-        dag: Dag,
-        run_directory: PathLike,
-        tdp_vars: PathLike,
+        operation_runner: OperationRunner,
+        lock_dir: Path,
     ):
-        self.dag = dag
-        self.run_directory = run_directory
-        self.tdp_vars = tdp_vars
-        self.process = None
-
-    def run_nodes(self, user, *args, **kwargs):
-        is_locked = DEPLOY_LOCK.acquire(blocking=False)
-        if not is_locked:
-            raise StillRunningException("Failed to lock for deployment")
-        try:
-            if self.process:
-                if self.process.is_alive():
-                    raise StillRunningException("run_nodes is still running")
-            parent_conn, child_conn = Pipe()
-            self.process = RunnerProcess(
-                self.dag,
-                self.run_directory,
-                self.tdp_vars,
-                user,
-                child_conn,  # type: ignore
-                *args,
-                **kwargs,
-            )
-            self.process.start()
-            result = parent_conn.recv()
-            parent_conn.close()
-            if result != 0:
-                raise StillRunningException(
-                    "RunnerProcess failed to lock for execution"
-                )
-        finally:
-            DEPLOY_LOCK.release()
-
-    @property
-    def running(self) -> bool:
-        return not self.process is None and self.process.is_alive()
-
+        self.operation_runner = operation_runner
+        self._thread_lock = Lock()
```

Comment:
Can you add comments to explain why we need 2 locks and if the `acquire` order is important ?

Reply:
Done

Reply:
Undone